pkg/k8s/container.go

```diff
@@ -0,0 +1,120 @@
+package k8s
+
+import (
+	"fmt"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Container struct {
+	ctr corev1.Container
+
+	srvc *corev1.Service
+
+	imageComponent components.Component
+
+	mountedSecrets    []*corev1.Secret
+	mountedConfigMaps []*corev1.ConfigMap
+}
+
+type VolumeMountOption func(*corev1.VolumeMount)
+
+func NewContainer(name string, imageComponent components.Component) *Container {
+	return &Container{
+		imageComponent: imageComponent,
+		ctr: corev1.Container{
+			Name:            name,
+			ImagePullPolicy: render.ImagePullPolicy(),
+		},
+	}
+}
+
+func (ctr *Container) ApplyImageSet(installation *operatorv1.InstallationSpec, imageSet *operatorv1.ImageSet) error {
+	reg := installation.Registry
+	path := installation.ImagePath
+	prefix := installation.ImagePrefix
+
+	image, err := components.GetReference(ctr.imageComponent, reg, path, prefix, imageSet)
+	if err != nil {
+		return err
+	}
+
+	ctr.ctr.Image = image
+	return nil
+}
+
+func (ctr *Container) AddEnv(env ...corev1.EnvVar) *Container {
+	ctr.ctr.Env = append(ctr.ctr.Env, env...)
```

Comment:
I think it would be worth a check that we don't have duplicate env vars, and a panic if we do.